### PR TITLE
syncer/: set dml worker timeout to 5 minutes

### PR DESCRIPTION
**Pull request by lichunzhu**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflow/issues/4894

### What is changed and how it works?
set timeout for dml execution from 30s to 5m.

### Check List 

Tests 

 - Unit test

Related changes

 - Need to cherry-pick to the release branch

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Review by lance6716 — APPROVED** [review]

File: dm/syncer/dml_worker.go [hunk 1]
```diff
@@ -249,7 +249,9 @@ func (w *DMLWorker) executeBatchJobs(queueID int, jobs []*job) {
 		time.Sleep(time.Duration(t) * time.Second)
 	})
 	// use background context to execute sqls as much as possible
-	ctx, cancel := w.syncCtx.WithTimeout(maxDMLExecutionDuration)
+	// set timeout to maxDMLConnectionDuration to make sure dmls can be replicated to downstream event if the latency is high
+	// if users need to quit this asap, we can support pause-task/stop-task --force in the future
+	ctx, cancel := w.syncCtx.WithTimeout(maxDMLConnectionDuration)
```
Comment (Ehco1996) [inline dm/syncer/dml_worker.go:254]: do we need to delete var `maxDMLExecutionDuration` ?

**Pull request merged**